src/main/java/com/pubnub/api/models/consumer/history/MessageType.java

```diff
@@ -0,0 +1,24 @@
+package com.pubnub.api.models.consumer.history;
+
+
+public class MessageType {
+    private PnMessageType pnMessageType;
+    private final String userMessageType;
+
+    public MessageType(String userMessageType) {
+        this.userMessageType = userMessageType;
+    }
+
+    MessageType(PnMessageType pnMessageType, String userMessageType) {
+        this.pnMessageType = pnMessageType;
+        this.userMessageType = userMessageType;
+    }
+
+    public String getType() {
```

Comment:
Shouldn't we mark this as `@NotNull`? That way you don't have to do this check:

```
 if (messageType != null && messageType.getType() != null) {
             params.put(MESSAGE_TYPE_QUERY_PARAMETER, messageType.getType());
         }
```

 and only this will be enough

``` if (messageType != null) {
             params.put(MESSAGE_TYPE_QUERY_PARAMETER, messageType.getType());
         }
```

Reply:
We could add similar `@NotNull` to SpaceId. I don't know why I didn't add it back then